src/taipy/core/scenario/_scenario_manager.py

```diff
@@ -244,7 +249,10 @@ def __get_config(scenario: Scenario):
     def _hard_delete(cls, scenario_id: ScenarioId):
         scenario = cls._get(scenario_id)
         if scenario.is_primary:
-            raise DeletingPrimaryScenario
+            if len(cls._get_all_by_cycle(scenario.cycle)) > 1:
+                raise DeletingPrimaryScenario
+            _CycleManagerFactory._build_manager()._hard_delete(scenario.cycle.id)
+            return
```

Comment:
what about using else here? :D

Reply:
I would prefer avoid using else since we already raise the exception here.

Reply:
raise an error will prevent you from running the rest so I'm not sure how it will affect it, and the else is better in meaning :/ but sure I'm ok with it :D